Add concurrent segment search documentation (#4990)
opensearch-project/documentation-website: _api-reference/index-apis/stats.md

```json
GET /testindex/_stats?level=shards
```
{% include copy-curl.html %}

> ## Concurrent segment search

Reviewer: These stats will also show up in [nodes-stats](https://opensearch.org/docs/2.9/api-reference/nodes-apis/nodes-stats/) as well as `_cat/nodes`, `_cat/shards`, `_cat/indices` but for the `_cat` APIs users can find it in the help section of the API.

Reviewer: @kolchfa-aws since all of these stats are gated by the concurrent search feature flag, I was thinking this information should be kept all within the `concurrent-segment-search.md` page too? I don't think we need to add these to the API pages until the feature is GA and the feature flag is removed, but it would be good to have the table below somewhere.

Author: Yes, the two options are:
- keep the APIs with the concurrent segment page
- put the APIs in their respective pages. 
It's a matter of opinion, but I think it makes it easier if the API changes are kept in the API pages. Users won't see concurrent-segment-related fields if they don't enable the feature, so I don't think there's any harm in keeping them in the API pages, especially because they are in a separate table. The danger with keeping them in the main page is that someone can just remove the experimental header and not add the API changes in the GA release.

Reviewer: > The danger with keeping them in the main page is that someone can just remove the experimental header and not add the API changes in the GA release.

Maybe we can track this is an issue somewhere - perhaps https://github.com/opensearch-project/documentation-website/issues/3662

Author: Added a comment so we can track

> Starting with OpenSearch 2.10, you can perform [concurrent segment search](search-plugins/concurrent-segment-search/), in which each shard-level request will search the segments in parallel during query phase. If you enable concurrent segment search, the Index Stats API response will contain several additional fields with statistics about slices (units of work executed by a thread). For more information about slices, see [Concurrent segment search]({{site.url}}{{site.baseurl}}/search-plugins/concurrent-segment-search#searching-segments-concurrently).

Reviewer: Small nit: These stats will show up if the feature flag is enabled for concurrent search, regardless of if the cluster/index settings for concurrent search are enabled.

> Starting with OpenSearch 2.10, you can perform [concurrent segment search]({{site.url}}{{site.baseurl}}/search-plugins/concurrent-segment-search/), in which each shard-level request will search the segments in parallel during query phase. If you [enable the experimental concurrent segment search feature flag]({{site.url}}{{site.baseurl}}/search-plugins/concurrent-segment-search#enabling-the-feature-flag), the Index Stats API response will contain several additional fields with statistics about slices (units of work executed by a thread). These fields will be provided whether or not the cluster and index settings for concurrent segment search are enabled. For more information about slices, see [Concurrent segment search]({{site.url}}{{site.baseurl}}/search-plugins/concurrent-segment-search#searching-segments-concurrently).

Reviewer 2: ```suggestion
Starting in OpenSearch 2.10, [concurrent segment search]({{site.url}}{{site.baseurl}}/search-plugins/concurrent-segment-search/) allows each shard-level request to search segments in parallel during the query phase. If you [enable the experimental concurrent segment search feature flag]({{site.url}}{{site.baseurl}}/search-plugins/concurrent-segment-search#enabling-the-feature-flag), the Index Stats API response will contain several additional fields with statistics about slices (units of work executed by a thread). These fields will be provided whether or not the cluster and index settings for concurrent segment search are enabled. For more information about slices, see [Concurrent segment search]({{site.url}}{{site.baseurl}}/search-plugins/concurrent-segment-search#searching-segments-concurrently).
```
